pkg/controller/plan/adapter/ova/ovfparser.go

```diff
@@ -60,3 +108,26 @@ func GetFirmwareFromConfig(vmConfigXML string) (firmware string, err error) {
 	}
 	return BIOS, nil
 }
+
+func GetOperationSystemFromConfig(vmConfigXML string) (os string, err error) {
+	xmlConf, err := readConfFromXML(vmConfigXML)
+	if err != nil {
+		return
+	}
+	return mapOs(xmlConf.Metadata.LibOsInfo.V2VOS.ID), nil
+}
+
+func mapOs(xmlOs string) (os string) {
+	split := strings.Split(xmlOs, "/")
+	distro := split[3]
+	if distro == "rocky" || distro == "opensuse" || distro == "ubuntu" || distro == "fedora" {
+		os = distro
+	} else {
+		os = split[3] + strings.Split(split[4], ".")[0]
+	}
+	os, ok := osV2VMap[os]
+	if !ok {
+		os = "otherGuest64"
```

Comment:
shall we log the `os` that was received?

Reply:
we can see it in the conversion pod and on the VM status the result. but yeah that maybe can help debugging.

Reply:
right, I meant only when we get to line 142 above so that if this happens, it will be more visible to us